DesktopUI2/DesktopUI2/Views/Windows/Dialogs/MappingViewDialog.xaml

```diff
@@ -0,0 +1,209 @@
+                  </TextBlock>
+                  <TextBlock
+                    Name="OutgoingType"
+                    IsVisible="{Binding OutgoingType, Converter={x:Static StringConverters.IsNotNullOrEmpty}}"
+                    Focusable="False"
+                    Margin="0"
+                    Grid.Column="1"
+                    Text="{Binding OutgoingType}"
+                    VerticalAlignment="Center">
+                  </TextBlock>
+                  <TextBlock
+                    IsVisible="{Binding OutgoingType, Converter={x:Static StringConverters.IsNullOrEmpty}}"
+                    Focusable="False"
+                    Margin="0"
+                    Grid.Column="1"
+                    Foreground="Gray"
+                    Text="{Binding InitialGuess}"
+                    VerticalAlignment="Center">
+                  </TextBlock>
+                </Grid>
+              </DataTemplate>
+            </ListBox.ItemTemplate>
+          </ListBox>
+        </StackPanel>
+        </Border>
+      </ScrollViewer>
+    </Grid>
+
+
+    <StackPanel
+      Background="{DynamicResource MaterialDesignCardBackground}"
+      Grid.Row="3"
+      Grid.ColumnSpan="3">
+      <Grid
+        ColumnDefinitions="*, auto"
+        RowDefinitions="auto, auto">
+        <StackPanel Orientation="Horizontal">
+          <TextBlock/>
+          <TextBlock/>
+        </StackPanel>
+
+        <!-- Search Box -->
+        <TextBox
+          Name="TypeSearchBox"
+          Grid.Column="0"
+          Grid.Row="1"
+          Margin="10"
+          IsEnabled="True"
+          Padding="0"
+          assists:TextFieldAssist.CornerRadius="32"
+          BorderBrush="{Binding $self.Foreground}"
+          BorderThickness="1"
+          Classes="Solo"
+          Text="{Binding SearchQuery, Mode=TwoWay}"
+          Watermark="Search Types in Project">
+          <TextBox.InnerLeftContent>
+            <Border Width="{Binding $self.Bounds.Height}" Height="{Binding $parent[TextBox].Height}">
+              <icons:MaterialIcon
+                Width="20"
+                Height="20"
+                HorizontalAlignment="Center"
+                VerticalAlignment="Center"
+                Kind="Magnify" />
+            </Border>
+          </TextBox.InnerLeftContent>
+        </TextBox>
+      </Grid>
+      <ScrollViewer
+        Grid.Row="2"
+        MaxHeight="110"
+        VerticalScrollBarVisibility="Auto">
+        <StackPanel>
+          <ListBox
+            Margin="0,0,0,8"
+            Classes="Chip"
+            Items="{Binding SearchResults}"
+            SelectedItem="{Binding SelectedType}">
+          </ListBox>
+        </StackPanel>
+      </ScrollViewer>
+      <StackPanel
+        Orientation="Horizontal"
+        HorizontalAlignment="Right">
+        <Button
+          MaxWidth="200"
+          CornerRadius="20"
+          Margin="5"
+          HorizontalAlignment="Left"
+          Command="{Binding ImportFamily}">
```

Comment:
Very minor remark: I would prefer commands to be named like `ImportFamilyCommand` so it's easier to understand they have a binding in the c# code and avoid the risk of deleting/renaming them involuntarily ;)